plone/namedfile/utils.py

```diff
@@ -77,3 +86,155 @@ def stream_data(file):
+
+
+def rotate_image(image_data, method=None, REQUEST=None):
+    """Rotate Image if it has Exif Orientation Informations other than 1.
+
+    Do not use PIL.Image.rotate function as this did not transpose the image,
+    rotate keeps the image width and height and rotates the image around a
+    central point. PIL.Image.transpose also changes Image Orientation.
+    """
+    if getattr(image_data, 'read', None):
+        img = PIL.Image.open(image_data)
+    else:
+        img = PIL.Image.open(StringIO(image_data))
+
+    if 'exif' in img.info:
+        exif_data = piexif.load(img.info['exif'])
+
+        if piexif.ImageIFD.Orientation in exif_data['0th']:
+            orientation = exif_data['0th'][piexif.ImageIFD.Orientation]
+    else:
+        width, height = im.size()
+        exif_data = {
+            '0th': {
+                piexif.ImageIFD.XResolution: (width, 1),
+                piexif.ImageIFD.YResolution: (height, 1),
+            }
+        }
+
+    if method is not None:
+        orientation = method
+
+    log.debug('Rotate image with input orientation: %s', orientation)
+
+    fmt = img.format
+    if orientation == 1:  # not transform necessary
+        #img = img
+        pass
+    elif orientation == 2:
+        img = img.transpose(PIL.Image.FLIP_LEFT_RIGHT)
+    elif orientation == 3:
+        img = img.transpose(PIL.Image.ROTATE_180)
+    elif orientation == 4:
+        img = img.transpose(PIL.Image.ROTATE_180).transpose(PIL.Image.FLIP_LEFT_RIGHT)
+    elif orientation == 5:
+        img = img.transpose(PIL.Image.ROTATE_270).transpose(PIL.Image.FLIP_LEFT_RIGHT)
+        exif_data['0th'][piexif.ImageIFD.XResolution], exif_data['0th'][piexif.ImageIFD.YResolution] = exif_data['0th'][piexif.ImageIFD.YResolution], exif_data['0th'][piexif.ImageIFD.XResolution]
+    elif orientation == 6:
+        img = img.transpose(PIL.Image.ROTATE_270)
+        exif_data['0th'][piexif.ImageIFD.XResolution], exif_data['0th'][piexif.ImageIFD.YResolution] = exif_data['0th'][piexif.ImageIFD.YResolution], exif_data['0th'][piexif.ImageIFD.XResolution]
+    elif orientation == 7:
+        img = img.transpose(PIL.Image.ROTATE_90).transpose(PIL.Image.FLIP_LEFT_RIGHT)
+        exif_data['0th'][piexif.ImageIFD.XResolution], exif_data['0th'][piexif.ImageIFD.YResolution] = exif_data['0th'][piexif.ImageIFD.YResolution], exif_data['0th'][piexif.ImageIFD.XResolution]
+    elif orientation == 8:
+        img = img.transpose(PIL.Image.ROTATE_90)
+        exif_data['0th'][piexif.ImageIFD.XResolution], exif_data['0th'][piexif.ImageIFD.YResolution] = exif_data['0th'][piexif.ImageIFD.YResolution], exif_data['0th'][piexif.ImageIFD.XResolution]
+
+    exif_data['0th'][piexif.ImageIFD.Orientation] = 1  # delete orientation
+    #del(exif_data['0th'][piexif.ImageIFD.Orientation])
+
+    try:
+        exif_bytes = piexif.dump(exif_data)
+    except:
+        del(exif_data['Exif'][41729])
+        # This Element piexif.ExifIFD.SceneType cause error on dump
```

Comment:
Following the examples above imho it would be cleaner to use the constant:

    del(exif_data['Exif'][piexif.ExifIFD.SceneType]